routing/DELEGATED_CONTENT_ROUTING_HTTP.md

```diff
@@ -0,0 +1,261 @@
+- The work for processing each provider record should be idempotent so that it can be retried without excessive cost in the case of full or partial failure of the request
+- Default limit of 100 keys per request
+- Implements pagination according to the Pagination section
+
+## Pagination
+
+APIs that return collections of results should support pagination as follows:
+
+- If there are more results, then a `NextPageToken` field should include an opaque string value, otherwise it should be undefined
+- The value of `NextPageToken` can be specified as the value of a `pageToken` query parameter to fetch the next page
+  - Character set is restricted to the regex `[a-zA-Z0-9-_.~]+`, since this is intended to be used in URLs
+- The client continues this process until `NextPageToken` is undefined or doesn't care to continue
+- A `pageLimit` query parameter specifies the maximum size of a single page
+
+### Implementation Notes
+
+Servers are required to return *at most* `pageLimit` results in a page. It is recommended for pages to be as dense as possible, but it is acceptable for them to return any number of items in the closed interval [0, pageLimit]. This is dependent on the capabilities of the backing database implementation. For example, a query specifying a `transfer` filter for a rare transfer protocol should not *require* the server to perform a very expensive database query for a single request. Instead, this is left to the server implementation to decide based on the constraints of the database.
+
+Implementations should encode into the token whatever information is necessary for fetching the next page. This could be a base32-encoded JSON object like `{"offset":3,"limit":10}`, an object ID of the last scanned item, etc.
+
+## Error Codes
+
+- `501`: must be returned if a method/path is not supported
+- `429`: may be returned to indicate to the caller that it is issuing requests too quickly
+- `400`: must be returned if an unknown path is requested
+
+## CORS and Web Browsers
+
+Browser interoperability requires implementations to support
+[CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS).
+
+JavaScript client running on a third-party Origin must be able to send HTTP
+request to the endpoints defined in this specification, and read the received
+values. This means HTTP server implementing this API must (1) support
+[CORS preflight requests](https://developer.mozilla.org/en-US/docs/Glossary/Preflight_request)
+sent as HTTP OPTIONS, and (2) always respond with headers that remove CORS
+limits, allowing every website to query the API for results:
+
+```plaintext
+Access-Control-Allow-Origin: *
+Access-Control-Allow-Methods: GET, PUT, POST, DELETE, OPTIONS
+```
+
+## Known Transfer Protocols
+This section contains a non-exhaustive list of known transfer protocols (by name) that may be supported by clients and servers.
+
+### Bitswap
+Multicodec name: `transport-bitswap`
+Schema: `bitswap`
```

Comment:
I feel this section needs explanation why we went with `transport-bitswap` and don't reuse  [libp2p identify's protocols list](https://github.com/libp2p/specs/tree/master/identify#protocols):

```console
$ ipfs id | jq .Protocols
[
  "/ipfs/bitswap",
  "/ipfs/bitswap/1.0.0",
  "/ipfs/bitswap/1.1.0",
  "/ipfs/bitswap/1.2.0",
  ...
]
```

Is this to facilitate non-libp2p transports, such as hypothetical `transport-http-gateway` which would have a list of gateways that are known to have specific CID or DAG? and support [verifiable retrieval](https://docs.ipfs.tech/reference/http/gateway/#trustless-verifiable-retrieval)? 

If so, I feel this spec would benefit from such an example, even if it is only aspirational.